shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresher.java

```diff
@@ -44,7 +45,7 @@ public void refresh(final ShardingSphereSchema schema, final Collection<String>
         if (containsInTableContainedRule(tableName, materials)) {
             tableMetaData = TableMetaDataBuilder.build(tableName, materials).orElse(new TableMetaData());
         } else {
-            tableMetaData = new TableMetaData();
+            tableMetaData = TableMetaDataLoader.load(materials.getDataSourceMap().get(routeDataSourceNames.iterator().next()), tableName, materials.getDatabaseType()).orElse(new TableMetaData());
```

Comment:
Is  it necessary to check `routeDataSourceNames.size()`

Reply:
The tables in DDL and not in table rule,  there should be only one route data source，may be we should check `routeDataSourceNames.size() !=0` ?